lms/djangoapps/certificates/migrations/0011_certificategenerationcoursesetting_can_self_generate.py

```diff
@@ -0,0 +1,25 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.apps import apps
+from django.db import migrations, models
+from django.db.models import F
+
+def copy_field(apps, schema_editor):
+    MyModel = apps.get_model('certificates', 'certificategenerationcoursesetting')
+    MyModel.objects.all().update(can_self_generate=F('enabled'))
+
+class Migration(migrations.Migration):
+    
+    dependencies = [
+        ('certificates', '0010_certificate_language_course_setting'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='certificategenerationcoursesetting',
+            name='can_self_generate',
+            field=models.BooleanField(default=False),
+        ),
+        migrations.RunPython(copy_field)
```

Comment:
Are there any race conditions here? IE when you run this, is it possible for someone to save a new row to the table with just `enabled` and not have it get copied over to `can_self_generate`?

Reply:
There is a (slim?) chance that someone may be saving a new row to the table with just enabled, if they did that after the migrations ran but before the code was deployed. In the follow up PR, which will delete the old 'enabled' column, I can copy any rows that were missed before deleting (if there were any).

Reply:
We should talk about the safest way to run this. I think there are a couple of different approaches that might work.